modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_page.ftl

```diff
@@ -172,40 +170,40 @@ public class Page<T> {
 			}
 			else if (Objects.equals(jsonParserFieldName, "facets")) {
 				if (jsonParserFieldValue != null) {
-					page.setFacets(
-						Stream.of(
-							toStrings((Object[])jsonParserFieldValue)
-						).map(
-							this::parseToMap
-						).map(
-							facets -> new Facet(
-								(String)facets.get("facetCriteria"),
-								Stream.of(
-									(Object[])facets.get("facetValues")
-								).map(
-									object -> (String)object
-								).map(
-									this::parseToMap
-								).map(
-									facetValues -> new Facet.FacetValue(Integer.valueOf((String)facetValues.get("numberOfOccurrences")), (String)facetValues.get("term"))
-								).collect(
-									Collectors.toList()
-								))
-						).collect(
-							Collectors.toList()
-						));
+					List<Facet> facets = new ArrayList<>();
+
+					for(String string : toStrings((Object[])jsonParserFieldValue)){
+						List<Facet.FacetValue> facetValues = new ArrayList<>();
+
+						Map<String, Object> jsonParserFieldValuesMap = parseToMap(string);
+
+						for(Object object : (Object[])jsonParserFieldValuesMap.get("facetValues")){
+							Map<String, Object> facetValueMap = parseToMap((String)object);
```

Comment:
@jeffwu0724 This class itself is a `PageJSONParser`, which is mutable (stateful) when parsing the json. Please double check, or maybe contact the corresponding product team, to see if it's correct to directly call `parseMap` in this branch ("facets"), while calling the method on a newly created instance in the branch above ("actions").

If it's OK to always directly call `this.parseToMap`, simplify the branch above.

Reply:
@dantewang , everytime when we call `parseToMap`, seems like it will init everything, and make sure the PageJSONParse is reset. Please see
'''
	private void _init(String json) {
		_captureStartStack = new Stack<>();
		_dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXX");
		_index = 0;
		_json = json.trim();
		_lastChar = 0;
	}
'''
In this case, I think it should be correct to call this.parseToMap directly, without declare a new one as what they did in ("action"). 
Please correct me if I am wrong. 
Thanks! :D

Reply:
@jeffwu0724 Since rest builder builds a lot of classes, let's confirm with the authors. If `parseToMap` can be used directly in the `"actions"` case, we can make this change.